va/va_test.go

```diff
@@ -671,7 +671,7 @@ func TestTLSSNI01TimeoutAfterConnect(t *testing.T) {
 		t.Fatalf("Connection should've timed out")
 	}
 	test.AssertEquals(t, prob.Type, probs.ConnectionProblem)
-	expected := "Timeout after connect (your server may be slow or overloaded)"
+	expected := "Timeout during read (your server may be slow or overloaded)"
```

Comment:
Can you update the va tests to include a server that sleeps during the `Handshake`? I remember you suggesting this when this was in my sprint. Something like overloading a callback for providing a cert and adding a long blocking sleep.

Reply:
I had forgotten that we already have a server like that in these unittests, and that's what's used in this particular test:

https://github.com/letsencrypt/boulder/blob/8167d877c35d7dd0ed77da73d9a02e4bf635410f/va/va_test.go#L632-L642

In other words, we were already testing that we do the right thing when we time out during a handshake, in terms of error messages to the user. An ideal test for this fix would check that we haven't leaked goroutines or TCP connections, but I'm not sure how to achieve that reliably. I'll take a quick look at what's available to us.

Reply:
> In other words, we were already testing that we do the right thing when we time out during a handshake, in terms of error messages to the user.

Ahhh! Ok that makes sense :+1: 

> An ideal test for this fix would check that we haven't leaked goroutines or TCP connections, but I'm not sure how to achieve that reliably. I'll take a quick look at what's available to us.

Knowing that the existing `slowTLSSrv` is there makes me think its fine to +1 this as-is if it turns out to be fiddly testing the leak.